lib/mpl_toolkits/mplot3d/proj3d.py

```diff
@@ -17,11 +17,13 @@ def _line2d_seg_dist(p1, p2, p0):
     and intersection point lies within segment if u is between 0 and 1
     """
 
-    x21 = p2[0] - p1[0]
-    y21 = p2[1] - p1[1]
     x01 = np.asarray(p0[0]) - p1[0]
     y01 = np.asarray(p0[1]) - p1[1]
+    if p1 == p2:  # Avoid RuntimeWarning
```

Comment:
This comment is kind of vague; what `RuntimeWarning`? You should say what is wrong conceptually here, not the effective resulting warning.

Reply:
New attempt.